ts/components/portfolio/PayLayout.tsx

```diff
@@ -0,0 +1,251 @@
+    return (
+      <Grid style={PortfolioStyles.topContainer}>>
+        <Col size={1}/>
+        <Col size={14}>
+          { this.topPortfolioTitle ()}
+          { this.topPortfolioSubtitle() }
+          { this.getTouchableContent() }
+        </Col>
+        <Col size={1}/>
+      </Grid>
+    );
+  }
+
+  private topPortfolioTitle(): React.ReactNode {
+
+    return (
+      <Row size={this.rowSize(PLC.TITLE_ROW)}>
+        <Grid>
+          <Col size={2}>
+            <Row size = {1}/>
+            <Row size={1}>
+              <H1 style={PortfolioStyles.pftitle}>{this.props.title}</H1>
+            </Row>
+          </Col>
+          { this.optionalRightImage() }
+        </Grid>
+      </Row>
+    );
+  }
+
+  private optionalRightImage(): React.ReactNode {
+    const images = {
+      [ImageType.BANK_IMAGE]: require("../../../img/portfolio/portfolio-icon.png")
+    };
+
+    if (this.props.rightImage !== undefined) {
+      return (
+        <Col size={1}>
+          <Image
+            source={images[this.props.rightImage]}
+            style={PortfolioStyles.pfImage}
+          />
+        </Col>
+      );
+    }
+    return null;
+  }
+
+  private topPortfolioSubtitle(): React.ReactNode {
+    if (this.rowSize(PLC.SUBTITLE_ROW) > 0) {
+      if (this.props.subtitle) {
+        // show single subtitle
+        return (
+          <Row size={this.rowSize(PLC.SUBTITLE_ROW)}>
+            <Grid>
+              <Row><Text style={PortfolioStyles.pfText}>{this.props.subtitle}</Text></Row>
+            </Grid>
+          </Row>
+        );
+      }
+      else if (this.props.subtitleLeft || this.props.subtitleRight) {
+        return (
+          <Row size={this.rowSize(PLC.SUBTITLE_ROW)}>
+            <Grid>
+              <Col size={1}>
+                <Row>
+                  <Left>
+                    <Text style={PortfolioStyles.pfSubtitleLeft}>{this.props.subtitleLeft}</Text>
+                  </Left>
+                </Row>
+              </Col>
+              <Col size={1}>
+                <Row>
+                  <Right>
+                    <Text
+                      style={PortfolioStyles.pfText}
+                      onPress={() =>
+                        // TODO: change to touchableContent
+                        this.props.navigation.navigate(ROUTES.PORTFOLIO_ADD_PAYMENT_METHOD) 
+                      }
+                    >
+                      {this.props.subtitleRight}
+                    </Text>
+                  </Right>
+                </Row>
+              </Col>
+            </Grid>
+          </Row>
+        );
+      }
+    }
+    return null;
+  }
+
+  private getTouchableContent(): React.ReactNode {
+    if (this.rowSize(PLC.TOUCHABLE_ROW) !== undefined) {
+      return (
+        <Row size={this.rowSize(PLC.TOUCHABLE_ROW)}>
+          {this.props.touchableContent}
+        </Row>
+      );
+    }
+    return null;
+  }
+  
+  private rowSize(rowIdentifier: PLC) : number {
```

Comment:
Sorry but this is really really hard to understand (and it's going to be hard to maintain) - it looks like that size in rows of each sub-component vary based on the below logic (that is imho a little obscure) that is in turn based on some combination of the component props. My suggestion is to make explicit the different "configurations" of this component, perhaps by making the props of the component an algebraic type.

```ts

type Configuration1 = {
  kind: "Configuration1",
  subtitleLeft: string
};

type Configuration2 = {
  kind: "Configuration2",
  subtitleRight: string
}

type Configuration = Configuration1 | Configuration2;

type Props = {
  title: string,
  configuration: Configuration
}

// then you can have a lookup map from props.configuration.kind to each subcomponent props

```